addons/website/static/src/js/widgets/link_popover_widget.js

```diff
@@ -64,53 +67,6 @@ const NavbarLinkPopoverWidget = weWidgets.LinkPopoverWidget.extend({
      * @param {Event} ev
      */
     _onEditLinkClick(ev) {
-        var self = this;
-        var $menu = this.$target.find('[data-oe-id]');
-        this.trigger_up('menu_dialog', {
-            name: $menu.text(),
-            url: $menu.parent().attr('href'),
-            save: (name, url) => {
-                let websiteId;
-                this.trigger_up('context_get', {
-                    callback: ctx => websiteId = ctx['website_id'],
-                });
-                const data = {
-                    id: $menu.data('oe-id'),
-                    name,
-                    url,
-                };
-                return this._rpc({
-                    model: 'website.menu',
-                    method: 'save',
-                    args: [websiteId, {'data': [data]}],
-                }).then(function () {
-                    self.options.wysiwyg.odooEditor.observerUnactive();
-                    self.$target.attr('href', url);
-                    $menu.text(name);
-                    self.options.wysiwyg.odooEditor.observerActive();
-                });
-            },
-        });
-        this.popover.hide();
-    },
-    /**
-     * Opens the menu tree editor. On menu editor save, current page changes
-     * will also be saved.
-     *
-     * @private
-     * @param {Event} ev
-     */
-     _onEditMenuClick(ev) {
-        const contentMenu = this.target.closest('[data-content_menu_id]');
-        const rootID = contentMenu ? parseInt(contentMenu.dataset.content_menu_id, 10) : undefined;
-        this.trigger_up('action_demand', {
-            actionName: 'edit_menu',
-            params: [rootID],
-        });
-    },
-});
-
-// Exact same static method but instantiating the specialized class.
-NavbarLinkPopoverWidget.createFor = weWidgets.LinkPopoverWidget.createFor;
-
-export default NavbarLinkPopoverWidget;
+        this.onEditLinkClick(this);
```

Comment:
I'm not sure I understand the point of this indirection. I'm probably missing something here. I guess the two functions are overridable and they just happen to be equivalent in this particular case ? ...but somehow the `onEditLinkClick` param does not seem to be used anywhere else than when the `NavbarLinkPopoverWidget` is instantiated so why bother ? Some reference issue that you encapsulate in a closure ?